src/os/bluestore/BlueStore.h

```diff
@@ -194,7 +200,7 @@ class BlueStore : public ObjectStore {
 
     Collection(BlueStore *ns, coll_t c);
   };
-  typedef ceph::shared_ptr<Collection> CollectionRef;
+  typedef boost::intrusive_ptr<Collection> CollectionRef;
```

Comment:
The internal Collection is a child of ObjectStore::CollectionImpl, so that 
we just need to cast it (no additional ptr indirection).